x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/Range.java

```diff
@@ -85,11 +85,7 @@ public ZoneId zoneId() {
 
     @Override
     public boolean foldable() {
-        if (lower.foldable() && upper.foldable()) {
-            return areBoundariesInvalid() || value.foldable();
```

Comment:
I think `areBoundariesInvalid` should be considered because it ensures that conditions like `WHERE foo BETWEEN 10 AND 1` will be reduced to `WHERE FALSE` which in turn can be optimized to a locally executed query not touching the index.

My suggestion would be to improve `areBoundariesInvalid` to only return `true` in the right cases. E.g. not if the range compares a date field to strings because the strings might be date math expression that need to be evaluated by ES.

Reply:
Thanks for your tips, I think it's more efficient  to only return `true` in the right cases for  `areBoundariesInvalid` method.    I have optimized the method  with a check of date field